databroker/databroker.py

```diff
@@ -500,8 +501,4 @@ def get_fields(header):
     -------
     fields : set
     """
-    fields = set()
-    for desc in header['descriptors']:
-        for field in desc['data_keys'].keys():
-            fields.add(field)
-    return fields
+    return set(chain(list(d['data_keys'].keys()) for d in header['descriptors']))
```

Comment:
@ericdill I think you're just missing a `*` there. Also, `list` is not necessary.

```python
In [13]: set(chain(*[[1, 2, 3], [2, 3, 4]]))
Out[13]: {1, 2, 3, 4}
```